pkg/networkserver/mac_device_mode.go

```diff
@@ -31,9 +31,15 @@ func handleDeviceModeInd(ctx context.Context, dev *ttnpb.EndDevice, pld *ttnpb.M
 		return nil, errNoPayload
 	}
 
-	dev.MACState.DeviceClass = pld.Class
+	switch {
+	case pld.Class == ttnpb.CLASS_C && dev.SupportsClassC:
+		dev.MACState.DeviceClass = ttnpb.CLASS_C
+
+	case pld.Class == ttnpb.CLASS_A:
+		dev.MACState.DeviceClass = ttnpb.CLASS_A
+	}
 	conf := &ttnpb.MACCommand_DeviceModeConf{
-		Class: pld.Class,
+		Class: dev.MACState.DeviceClass,
 	}
```

Comment:
No class B here?

Reply:
No, but it's an extremely small change. I was working on class transitioning and noticed this. I can extract this into a separate PR if you want (see OP)